front/lib/models.ts

```diff
@@ -1241,6 +1241,194 @@ DataSource.hasMany(DocumentTrackerChangeSuggestion, {
   onDelete: "CASCADE",
 });
 
+// Assistant V2
+export class Conversation extends Model<
+  InferAttributes<Conversation>,
+  InferCreationAttributes<Conversation>
+> {
+  declare id: number;
+  declare title: string;
+  declare created: Date;
+  declare visibility: string;
+}
+
+Conversation.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    title: {
+      type: DataTypes.TEXT,
+      allowNull: false,
+    },
+    created: {
+      type: DataTypes.DATE,
+      allowNull: false,
+    },
+    visibility: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+  },
+  {
+    modelName: "conversation",
+    sequelize: front_sequelize,
+  }
+);
+
+export class UserMessage extends Model<
+  InferAttributes<UserMessage>,
+  InferCreationAttributes<UserMessage>
+> {
+  declare id: number;
+  declare textContent: string;
+}
+
+UserMessage.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    textContent: {
+      type: DataTypes.TEXT,
+      allowNull: false,
+    },
+  },
+  {
+    modelName: "user_message",
+    sequelize: front_sequelize,
+  }
+);
+
+export class AssistantMessage extends Model<
+  InferAttributes<AssistantMessage>,
+  InferCreationAttributes<AssistantMessage>
+> {
+  declare id: number;
+  declare textContent: string | null;
+}
+
+AssistantMessage.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    textContent: {
+      type: DataTypes.TEXT,
+      allowNull: true,
+    },
+  },
+  {
+    modelName: "assistant_message",
+    sequelize: front_sequelize,
+  }
+);
+
+export class Message extends Model<
+  InferAttributes<Message>,
+  InferCreationAttributes<Message>
+> {
+  declare id: number;
+  declare parentId: number | null;
+  declare userMessageId: number | null;
+  declare assistantMessageId: number | null;
+  declare version: number;
+  declare is_deleted: boolean;
+  declare conversation_id: number;
+  declare rank: number;
+}
+
+Message.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    parentId: {
```

Comment:
@spolu  Didn't make this unique, so making the assumption that multiple messages can have the same parent. WDYT ?